pandas/_libs/src/vendored/numpy/datetime/np_datetime.c

```diff
@@ -482,10 +482,20 @@ npy_datetime npy_datetimestruct_to_datetime(NPY_DATETIMEUNIT base,
 
   if (base == NPY_FR_ns) {
     int64_t nanoseconds;
-    PD_CHECK_OVERFLOW(
-        scaleMicrosecondsToNanoseconds(microseconds, &nanoseconds));
-    PD_CHECK_OVERFLOW(
-        checked_int64_add(nanoseconds, dts->ps / 1000, &nanoseconds));
+
+    // for near-minimum timestamps, scaling microseconds to nanoseconds
+    // overflows but adding nanoseconds puts the timestamp back in a valid range
+    const int64_t min_nanoseconds = NPY_MIN_INT64 + 1;
+    if (microseconds == min_nanoseconds / 1000 - 1) {
+      // calculate final nanoseconds from minimum without scaling microseconds
+      PD_CHECK_OVERFLOW(checked_int64_add(
```

Comment:
I don't think overflow is possible here so can remove the checks

Reply:
This will overflow when `dts->ps < _NS_MIN_DTS.ps`.

This is the case for the following timestamps: `1677-09-21 00:12:43.145224000` - `1677-09-21 00:12:43.145224192`, because then `dts->ps` is in `[0, 192]`, but `_NS_MIN_DTS.ps` is `193`, so we'd add negative nanoseconds to the negative minimum, which should overflow.

For timestamps below `1677-09-21 00:12:43.145224000`, the above condition wouldn't trigger (`microseconds == min_nanoseconds / 1000 - 1`), and it'd run into the overflow in the `else`.